llvm/test/tools/llvm-objcopy/DXContainer/extract-section-basic.test

```diff
@@ -0,0 +1,320 @@
+## Tests that a separate DXContainer is created for the RTS0 (root signature)
+## part, when--extract-section is specified.
+
+# RUN: yaml2obj %s -o %t
+# RUN: llvm-objcopy %t --extract-section=RTS0=%t.rts0.out
+# RUN: obj2yaml %t.rts0.out | FileCheck %s
+
+## The DXContainer described below was generated with:
+
+## `clang-dxc -T cs_6_7 test.hlsl /Fo temp.dxo`
+## `obj2yaml temp.dxo`
+
+## ``` test.hlsl
+## [RootSignature("")]
+## [numthreads(1,1,1)]
+## void main() {}
+## ```
+
+# CHECK:      Header:
+# CHECK-NEXT:   Hash:
+# CHECK:        Version:
+# CHECK-NEXT:      Major:           1
+# CHECK-NEXT:      Minor:           0
+# CHECK-NEXT:   FileSize:       68
+# CHECK-NEXT:   PartCount:     1
+# CHECK-NEXT:   PartOffsets:   [ 36 ]
+# CHECK-NEXT:   Parts:
+# CHECK-NOT: DXIL
+# CHECK-NOT: SFI0
+# CHECK-NOT: HASH
+# CHECK-NOT: ISG1
+# CHECK-NOT: OSG1
```

Comment:
Technically these sections aren't actually ordered, though in practice I guess they'll always happen to be in this order so it's probably fine. Could we do something like `CHECK-NOT: - Name: {{DXIL|SFI0|HASH|ISG1|OSG1|PSV0}}` to be more robust against ordering though? We could then repeat this same check-not after the RTS0 checks.